app/models/user.rb

```diff
@@ -1,9 +1,16 @@
 class User < ApplicationRecord
   # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  # :confirmable, :lockable, :timeoutable, :trackable, and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
-  has_many :participations
   has_many :classrooms, dependent: :destroy
-  has_many :classrooms_as_student, through: :participations, source: :classroom
+  has_many :participations, dependent: :destroy
+  has_many :students, through: :participations, source: :user
```

Comment:
Same here, did you create a reference anywhere else that if user is not a teacher then it is automatically a student for Ruby to understand? As that rule needs to work for other call methods as well and not to a specific controller/policy

Reply:
Im working on it rn

Reply:
so in our User model, I added this method called student? that basically does a quick check to see if a user isn't a teacher. It's like a handy tool you can use anywhere in your app to figure out if someone is a student without having to create a whole new reference or anything. So, whenever you need to know if a user is 
 the student, just do student?  and it'll tell you its student or not

Reply:
def student?
  !teacher?
end 
this should be added to user.rb
and its can be called anywhere like that
def some_action
    if current_user.student?
    else